pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2826,13 +2834,16 @@ public void readStartupMessages() throws IOException, SQLException {
         case 'K':
           // BackendKeyData
           int msgLen = pgStream.receiveInteger4();
-          if (msgLen != 12) {
-            throw new PSQLException(GT.tr("Protocol error.  Session setup failed."),
-                PSQLState.PROTOCOL_VIOLATION);
-          }
-
           int pid = pgStream.receiveInteger4();
-          int ckey = pgStream.receiveInteger4();
+          int keyLen = msgLen - 8;
+          byte[] ckey = new byte[keyLen];
```

Comment:
```suggestion
          byte[] ckey;
```

The allocation here is not needed. `pgStream.receive(..)` below would reallocate